src/git-town-prepend

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
+
+
+function ensure_has_target_branch {
+  if [ -z "$target_branch_name" ]; then
+    echo_error_header
+    echo_error "No branch name provided."
+    exit_with_error newline
+  fi
+}
+
+
+function preconditions {
+  target_branch_name=$1
+  ensure_has_target_branch
+
+  child_branch_name=$INITIAL_BRANCH_NAME
+  ensure_is_feature_branch "$INITIAL_BRANCH_NAME" "Only feature branches can have parent branches."
+
+  parent_branch_name=$(parent_branch "$child_branch_name")
+  if [ "$parent_branch_name" != "$MAIN_BRANCH_NAME" ]; then
+    ensure_knows_parent_branches "$parent_branch_name"
+  fi
```

Comment:
Good catch. This is much better logic. I just took what `git hack` had and stripped out  unnecessary stuff, leaving this in an imperfect state.